Lab3/client/Application/src/app/nav/nav.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, OnInit } from '@angular/core';
+import { BreakpointObserver, Breakpoints } from '@angular/cdk/layout';
+import { from, Observable, of } from 'rxjs';
+import { filter, map, shareReplay } from 'rxjs/operators';
+import {
+  NavigationCancel,
+  NavigationEnd,
+  NavigationError,
+  NavigationStart,
+  Router,
+} from '@angular/router';
+
+import { AuthenticatorService } from '@aws-amplify/ui-angular';
+import { Auth } from 'aws-amplify';
+import { navItems } from '../_nav';
+import { AuthConfigurationService } from './../views/auth/auth-configuration.service';
+
+@Component({
+  selector: 'app-nav',
+  templateUrl: './nav.component.html',
+  styleUrls: ['./nav.component.scss'],
+})
+export class NavComponent implements OnInit {
+  loading$: Observable<boolean> = of(false);
+  isAuthenticated$: Observable<Boolean> | undefined;
+  username$: Observable<string> | undefined;
+  companyName$: Observable<string> | undefined;
+  public navItems = navItems;
+  isHandset$: Observable<boolean> = this.breakpointObserver
+    .observe(Breakpoints.Handset)
+    .pipe(
+      map((result) => result.matches),
+      shareReplay()
+    );
+
+  constructor(
+    private breakpointObserver: BreakpointObserver,
+    private router: Router,
+    private authConfigService: AuthConfigurationService
+  ) {
+    // this.configSvc.loadConfigurations().subscribe((val) => console.log(val));
+    this.loading$ = this.router.events.pipe(
+      filter(
+        (e) =>
+          e instanceof NavigationStart ||
+          e instanceof NavigationEnd ||
+          e instanceof NavigationCancel ||
+          e instanceof NavigationError
+      ),
+      map((e) => e instanceof NavigationStart)
+    );
+  }
+
+  ngOnInit(): void {
+    if (!localStorage.getItem('tenantName')) {
```

Comment:
If tenantName is not set you are assuming that user in not authenticated right? Can you just check this using some amplify module?

Reply:
At that point, we can't really assume that amplify has been configured. So, all I do is check if for the tenantName in the localStorage. If it's not set, it's a good sign that amplify has not been configured. If it does exist, then we configure Amplify and use that to check if the user is logged in.